src/Features/CSharp/Portable/CodeFixes/AddParameterXmlDocumentation/AddParameterXmlDocumentationCodeFixProvider.cs

```diff
@@ -0,0 +1,316 @@
+
+                XmlNameAttributeSyntax adjacentAttribute;
+                bool adjacentAttributeIsBefore;
+                FindAdjacentParameterDocumentation(
+                    parameterName,
+                    allParameterNames: parameterList.Parameters.Select(p => p.Identifier.ValueText),
+                    tokenWithDocumentationComment: parameterList.Parent.GetFirstToken(),
+                    elementKind: XmlNameAttributeElementKind.TypeParameter,
+                    adjacentNameAttribute: out adjacentAttribute,
+                    adjacentAttributeIsBefore: out adjacentAttributeIsBefore);
+
+                if (adjacentAttribute == null) { return; }
+
+                context.RegisterCodeFix(
+                    new AddParameterXmlDocumentationCodeAction(context.Document, parameterName, adjacentAttribute,
+                        adjacentAttributeIsBefore, isTypeParameter: true),
+                    diagnostic);
+            }
+        }
+
+        /// <summary>
+        /// Finds a xml documentation tag with a name attribute at which another a new tag can be inserted.
+        /// </summary>
+        /// <param name="parameterName">The parameter of the new tag</param>
+        /// <param name="allParameterNames">All the parameter of which <paramref name="parameterName"/> is one.</param>
+        /// <param name="tokenWithDocumentationComment"></param>
+        /// <param name="elementKind">Specifies which elements can be found.</param>
+        /// <param name="adjacentNameAttribute">The name attribute of the adjacent parameter. If null no parameter was found.</param>
+        /// <param name="adjacentAttributeIsBefore">Indicates if the found adjacent attribute is before or after the parameter.</param>
+        private static void FindAdjacentParameterDocumentation(string parameterName, IEnumerable<string> allParameterNames, SyntaxToken tokenWithDocumentationComment,
+            XmlNameAttributeElementKind elementKind, out XmlNameAttributeSyntax adjacentNameAttribute, out bool adjacentAttributeIsBefore)
+        {
+            FindAdjacentParameterDocumentationWalker.FindAdjacentParameterDocumenteden(parameterName, allParameterNames, tokenWithDocumentationComment,
+                elementKind, out adjacentNameAttribute, out adjacentAttributeIsBefore);
+        }
+
+        private class FindAdjacentParameterDocumentationWalker : CSharpSyntaxWalker
+        {
+            private FindAdjacentParameterDocumentationWalker(int parameterIndex, Dictionary<string, int> parameterIndices, XmlNameAttributeElementKind elementKind)
+                : base(SyntaxWalkerDepth.StructuredTrivia)
+            {
+                _parameterIndex = parameterIndex;
+                _parameterIndices = parameterIndices;
+                _elementKind = elementKind;
+            }
+
+            private int _parameterIndex = -1;
+            private Dictionary<string, int> _parameterIndices = new Dictionary<string, int>();
```

Comment:
Why are you allocating a Dictionary here only to assign it with a different dictionary in the constructor?